DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/AnonymizerEngine.cs

```diff
@@ -0,0 +1,83 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Dicom;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Anonymizer.Core.AnonymizerConfigurations;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.Anonymizer.Core.Rules;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core
+{
+    public class AnonymizerEngine
+    {
+        private readonly ILogger _logger = AnonymizerLogging.CreateLogger<AnonymizerEngine>();
+        private readonly AnonymizerSettings _anonymizerSettings;
+        private readonly AnonymizerConfigurationManager _configurationManager;
+        private readonly IAnonymizerRuleFactory _ruleFactory;
+
+        public AnonymizerEngine(string configFilePath = "configuration.json", AnonymizerSettings anonymizerSettings = null, IAnonymizerRuleFactory ruleFactory = null)
+            : this(AnonymizerConfigurationManager.CreateFromConfigurationFile(configFilePath), anonymizerSettings, ruleFactory)
+        {
+        }
+
+        public AnonymizerEngine(AnonymizerConfigurationManager configurationManager, AnonymizerSettings anonymizerSettings = null, IAnonymizerRuleFactory ruleFactory = null)
+        {
+            EnsureArg.IsNotNull(configurationManager, nameof(configurationManager));
+
+            _anonymizerSettings = anonymizerSettings ?? new AnonymizerSettings();
+            _configurationManager = configurationManager;
+            _ruleFactory = ruleFactory ?? new AnonymizerRuleFactory(_configurationManager.Configuration, new DicomProcessorFactory(), new AnonymizerSettingsFactory());
```

Comment:
>AnonymizerSettingsFactory [](http://example.com/codeflow?start=138&length=25)

there are many similiar names here. for example, what's the difference and connections between AnonymizerSettingsFactory and AnonymizerSettings at line 32? Maybe you need to change some names to make it more accurate and easy to understand.

Reply:
Rename "AnonymizerSettings" to "AnonymizerEngineOptions". Settings are used for processors.